app.js

```diff
@@ -2,11 +2,15 @@ const express = require('express');
 const app = express();
 app.use(express.json())
 const { fetchAllTopics } = require('./controllers/controllers')
-const { handles500Errors, handles400Errors, handle404nonExistentPath } = require('./handlesErrors')
+const { handles500Errors, handles400Errors } = require('./handlesErrors')
 
 app.get("/api/topics", fetchAllTopics)
 
-app.use(handle404nonExistentPath)
+
+app.use((req, res, next) => {
+    res.status(404).send({ msg: 'Path not found' })
```

Comment:
The last thing you need to do is replace `app.use...` with `app.all("*",...` 

`app.use` will invoke in the order that the app.use is written while `app.all` is treated as part of the router like app.get, because of this app.use may cause you some issue with error handling later.